### PR TITLE
Replace golint with revive

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

- Use [revive](https://github.com/mgechev/revive) instead of [golint](https://github.com/golang/lint) as coding style checker.
- Fix issues found by revive.

### Motivation

Golint was [deprecated](https://github.com/golang/go/issues/38968). Revive is is a maintained replacement for it.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]
LGTM, I think you have to log in again on CircleCI so that it runs (and maybe you will have to change licenses because of that)

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/trace/stats/client_stats_aggregator_test.go [hunk 1]
```diff
@@ -290,8 +290,8 @@ func TestCountAggregation(t *testing.T) {
 			c1 := payloadWithCounts(testTime, tc.k, 11, 7, 100)
 			c2 := payloadWithCounts(testTime, tc.k, 27, 2, 300)
 			c3 := payloadWithCounts(testTime, tc.k, 5, 10, 3)
-			kDefault := bucketAggregationKey{}
-			cDefault := payloadWithCounts(testTime, kDefault, 0, 2, 4)
+			keyDefault := bucketAggregationKey{}
```
Comment (gbbr) [inline pkg/trace/stats/client_stats_aggregator_test.go:293]: Just out of curiosity - why?

Comment (KSerrania) [reply pkg/trace/stats/client_stats_aggregator_test.go:293]: `revive` sees `kDefault` as Hungarian notation for defining a constant, which it doesn't accept (see: https://github.com/golang/lint/issues/258, https://github.com/mgechev/revive/blob/5b4bd9a3c2b694d5030b8bc6fa45560fc1b600e1/rule/var-naming.go#L90-L98) and wants it to be replaced by `Default`. That's why it was replaced by the more explicit `keyDefault`, which doesn't trigger this rule.

**Review by mx-psi — APPROVED** [review]
Approving also for changes in `*.go` files, everything looks good

**Review by ahmed-mez — APPROVED** [review]

**Pull request merged**